Servers/SystemServer/main.cpp

```diff
@@ -0,0 +1,61 @@
+#include <stdio.h>
+#include <sys/types.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include <sched.h>
+
+void start_process(const char *prog, int prio)
+{
+    pid_t pid = 0;
+
+    while (true) {
+        dbgprintf("Forking for %s...\n", prog);
+        int pid = fork();
+        if (pid < 0) {
+            dbgprintf("Fork %s failed! %s\n", prog, strerror(errno));
+            continue;
+        } else if (pid > 0) {
+            // parent...
+            dbgprintf("Process %s hopefully started with priority %d...\n", prog, prio);
+            return;
+        }
+        break;
+    }
+
+    while (true) {
+        if (pid == 0) {
+            dbgprintf("Executing for %s... at prio %d\n", prog, prio);
+            struct sched_param p;
+            p.sched_priority = prio;
+            sched_setparam(pid, &p);
+
+            char* progv[256];
+            progv[0] = (char*)prog;
+            progv[1] = nullptr;
+            int ret = execv(prog, progv);
+            if (ret < 0) {
+                dbgprintf("Exec %s failed! %s", prog, strerror(errno));
+                continue;
+            }
+            break;
+        } else {
+            break;
+        }
+    }
+}
+
+int main(int, char **)
+{
+    int lowest_prio = sched_get_priority_min(SCHED_OTHER);
+    int highest_prio = sched_get_priority_max(SCHED_OTHER);
+    start_process("/bin/LookupServer", lowest_prio);
+    start_process("/bin/WindowServer", highest_prio);
+    start_process("/bin/Taskbar", highest_prio);
+    start_process("/bin/Terminal", highest_prio-1);
```

Comment:
A further expansion piece here would be to get Launcher to start things with priority highest-1.

Logic here being: background processes use the lowest priority, interactive ones use second highest, session processes use the highest.

Reply:
This makes sense, I could make a quick PR for that probably. Speaking of which, maybe the Launcher should handle starting the "default" apps (like the Terminal in this case). i.e., it seems odd for the OS to be what's handling starting up user-facing apps.

Reply:
You won't be able to PR it until this goes in (as this adds priority setting) - but other than that, sure :)

I think it's fine to leave terminal in SystemServer for the time being as it provides a nice, minimal place to just dump things in for testing. Eventually, yeah, they don't belong in that form, but that's further down the line when we have user sessions.